File: problems/subsets/backtracking.cs
// Solution using backtracking
public class Solution {
    public IList<IList<int>> Subsets(int[] nums) {
        var result = new List<IList<int>>();

        Backtrack(0, nums, new List<int>(), result);

        return result;
    }

    private void Backtrack(int start, int[] nums, List<int> subset, List<IList<int>> results)
    {
        results.Add(new List<int>(subset));

        for (var i = start; i < nums.Length; i++)
        {
            subset.Add(nums[i]);

            Backtrack(i + 1, nums, subset, results);

            // Remove last element
            subset.RemoveAt(subset.Count - 1);
        }
    }
}

File: problems/subsets/backtracking.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        results = []
        subset = []

        self.backtrack(0, nums, subset, results)

        return results

    def backtrack(self, start: int, nums: List[int], subset: List[int], results: List[List[int]]):
        results.append(subset[:])

        for i in range(start, len(nums)):
            subset.append(nums[i])

            self.backtrack(i + 1, nums, subset, results)

            subset.pop()